invenio/templates/configurations/logstash.yaml

```diff
@@ -52,67 +52,133 @@ data:
-        ##############################################################
-        # add ua/device only if set
-        if [@metadata][ua_device] {
-              mutate {
-                  add_field => { "[data][ua/device]" => "%{[@metadata][ua_device]}" }
-              }
-          }
-        # add ua/family only if set
-        if [@metadata][ua_name] {
-            mutate {
-                add_field => { "[data][ua/family]" => "%{[@metadata][ua_name]}" }
-            }
+        # Enrich with geolocation data.
+        geoip {
+            source => "http_x_forwarded_for" #field name that contains the ip address
+            fields => [ "city_name", "country_name", "country_code2", "continent_code"]
+            target => "data"
+            # Download static database and point to it
         }
-        # add ua/patch only if set
-        if [@metadata][ua_patch] {
-            mutate {
-                add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}.%{[@metadata][ua_patch]}" }
-            }
-        } else {
+
+      useragent {
+          source => "[data][user_agent]"
+          target => "[@metadata]"
+      }
+      ##############################################################
+      # USER AGENT
+      ##############################################################
+      # add ua/device only if set
+      if [@metadata][ua_device] {
             mutate {
-                add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}" }
+                add_field => { "[data][ua/device]" => "%{[@metadata][ua_device]}" }
             }
         }
-        # add ua/os only if set
-        if [@metadata][ua_os] {
-            mutate {
-                add_field => { "[data][ua/os]" => "%{[@metadata][ua_os]}" }
-            }
+      # add ua/family only if set
+      if [@metadata][ua_name] {
+          mutate {
+              add_field => { "[data][ua/family]" => "%{[@metadata][ua_name]}" }
+          }
+      }
+      # add ua/patch only if set
+      if [@metadata][ua_patch] {
+          mutate {
+              add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}.%{[@metadata][ua_patch]}" }
+          }
+      } else {
+          mutate {
+              add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}" }
+          }
+      }
+      # add ua/os only if set
+      if [@metadata][ua_os] {
+          mutate {
+              add_field => { "[data][ua/os]" => "%{[@metadata][ua_os]}" }
+          }
+      }
+      # add a field with the category, based on the url
+      grok {
+         match => { "url" => ["^/(api/)?records?/%{DATA}/(media-)?files/"]}
+         add_field => {"[data][request_category]" => "files" }
+      }
+      # add a field with the category "legacy", based on the url
+      grok {
```

Comment:
what about adding to this grok `tag_on_failure => [] ` . This will avoid adding the tag